hazelcast/src/test/java/com/hazelcast/test/HazelcastParallelClassRunner.java

```diff
@@ -29,7 +29,7 @@
 public class HazelcastParallelClassRunner extends AbstractHazelcastClassRunner {
 
     private static final int MAX_THREADS = !TestEnvironment.isMockNetwork() ? 1
-                : Math.max(Runtime.getRuntime().availableProcessors(), 8);
+                : Math.min(Runtime.getRuntime().availableProcessors()/2, 1);
```

Comment:
Doesn't it select `0` (if `availableProcessors()` is `1`) or `1` always? Because you have changed `Math.max` to `Math.min`.
In this case (`MAX_THREADS` is `1`),  `HazelcastParallelClassRunner` will not behave as parallel runner but serial runner.